### PR TITLE
chore: considers ignored namespaces in filters

### LINKED ISSUE #1162: ignoredNamespaces are not considered in filters (shouldSkip and filterMatch)
### Describe what should be investigated or refactored

We need to consider ignored namespaces in our filter functions. We have a static check at built time for check if namespace filters but if a module author is not using bindings and watching all pods for example, then the callback will still be processed.  

### Links to any relevant code
(optional) i.e. - https://github.com/defenseunicorns/pepr/blob/24f1f253b1db4c9d97717c0282b8f39c74d3ace1/README.md?plain=1#L3

### Additional context
Add any other context or screenshots about the technical debt here.

### PR DESCRIPTION
**Pull request by cmwylie19**
## Description

Considers ignored namespaces in filters when determining whether to process a workload.

Depends on: https://github.com/defenseunicorns/pepr/pull/1156

End to End Test:  https://github.com/defenseunicorns/pepr-excellent-examples/pull/75 specifically hello-pepr-regex-name

## Related Issue

Fixes #1162 

Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging
- [x] Unit, [Journey](https://github.com/defenseunicorns/pepr/tree/main/journey), [E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples), [docs](https://github.com/defenseunicorns/pepr/tree/main/docs), [adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or updated as needed
- [x] [Contributor Guide Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request) followed

### CODE REVIEW

File: src/lib/mutate-request.ts [hunk 1]
```diff
@@ -3,9 +3,8 @@
 
 import { KubernetesObject } from "kubernetes-fluent-client";
 import { clone, mergeDeepRight } from "ramda";
-
-import { AdmissionRequest, Operation } from "./k8s";
-import { DeepPartial } from "./types";
+import { Operation, AdmissionRequest, DeepPartial } from "./types";
+import {} from "./types";
```
Comment (cmwylie19) [inline src/lib/mutate-request.ts]:
```suggestion
```

File: src/lib/finalizer.test.ts [hunk 2]
```diff
@@ -5,7 +5,8 @@ import { beforeEach, describe, expect, it, jest } from "@jest/globals";
 import { addFinalizer, removeFinalizer } from "./finalizer";
 import { KubernetesObject, K8s, GenericClass, RegisterKind } from "kubernetes-fluent-client";
 import { K8sInit } from "kubernetes-fluent-client/dist/fluent/types";
-import { Operation, AdmissionRequest } from "./k8s";
+import { AdmissionRequest } from "./types";
+import { Operation } from "./types";
```
Comment (btlghrants) [inline src/lib/finalizer.test.ts]:
```suggestion
import { AdmissionRequest, Operation } from "./types";
```

File: src/lib/assets/index.ts [hunk 3]
```diff
@@ -65,7 +65,7 @@ export class Assets {
     this.capabilities = await loadCapabilities(this.path);
     // give error if namespaces are not respected
     for (const capability of this.capabilities) {
-      namespaceComplianceValidator(capability, this.alwaysIgnore.namespaces);
+      namespaceComplianceValidator(capability, this.alwaysIgnore?.namespaces || []);
```
Comment (cmwylie19) [inline src/lib/assets/index.ts]:
```suggestion
      namespaceComplianceValidator(capability, this.alwaysIgnore?.namespaces);
```

File: src/lib/assets/webhooks.ts [hunk 4]
```diff
@@ -82,7 +82,7 @@ export async function webhookConfig(
   const ignore = [peprIgnoreLabel];
 
   const { name, tls, config, apiToken, host } = assets;
-  const ignoreNS = concat(peprIgnoreNamespaces, config.alwaysIgnore.namespaces || []);
+  const ignoreNS = concat(peprIgnoreNamespaces, config?.alwaysIgnore?.namespaces || []);
```
Comment (cmwylie19) [inline src/lib/assets/webhooks.ts:85]:
```suggestion
  const ignoreNS = concat(peprIgnoreNamespaces, config?.alwaysIgnore?.namespaces);
```

File: src/lib/module.ts [hunk 5]
```diff
@@ -108,7 +108,7 @@ export class PeprModule {
       // Wait for the controller to be ready before setting up watches
       if (isWatchMode() || isDevMode()) {
         try {
-          setupWatch(capabilities);
+          setupWatch(capabilities, pepr?.alwaysIgnore?.namespaces || []);
```
Comment (cmwylie19) [inline src/lib/module.ts]:
```suggestion
          setupWatch(capabilities, pepr?.alwaysIgnore?.namespaces);
```

File: src/lib/mutate-processor.ts [hunk 6]
```diff
@@ -49,7 +50,7 @@ export async function mutateProcessor(
       }
 
       // Continue to the next action without doing anything if this one should be skipped
-      if (shouldSkipRequestRegex(action, req, namespaces)) {
+      if (shouldSkipRequestRegex(action, req, namespaces, config?.alwaysIgnore?.namespaces || [])) {
```
Comment (cmwylie19) [inline src/lib/mutate-processor.ts]:
```suggestion
      if (shouldSkipRequestRegex(action, req, namespaces, config?.alwaysIgnore?.namespaces)) {
```

File: src/lib/validate-request.test.ts [hunk 7]
```diff
@@ -3,11 +3,9 @@
 
 import { beforeEach, describe, expect, it } from "@jest/globals";
 import { KubernetesObject } from "kubernetes-fluent-client";
-
-import { Operation, AdmissionRequest } from "./k8s";
-import { ValidateActionResponse } from "./types";
+import { ValidateActionResponse, AdmissionRequest } from "./types";
```
Comment (btlghrants) [inline src/lib/validate-request.test.ts]:
```suggestion
import { ValidateActionResponse, AdmissionRequest, Operation } from "./types";
```

File: src/lib/validate-request.test.ts [hunk 8]
```diff
@@ -3,11 +3,9 @@
...
 import { PeprValidateRequest } from "./validate-request";
-
+import { Operation } from "./types";
```
Comment (btlghrants) [inline src/lib/validate-request.test.ts]:
```suggestion
```

File: src/lib/validate-processor.ts [hunk 9]
```diff
@@ -41,7 +44,7 @@ export async function validateProcessor(
       };
 
       // Continue to the next action without doing anything if this one should be skipped
-      if (shouldSkipRequestRegex(action, req, namespaces)) {
+      if (shouldSkipRequestRegex(action, req, namespaces, config?.alwaysIgnore?.namespaces || [])) {
```
Comment (cmwylie19) [inline src/lib/validate-processor.ts]:
```suggestion
      if (shouldSkipRequestRegex(action, req, namespaces, config?.alwaysIgnore?.namespaces)) {
```

**Review by btlghrants — APPROVED** [review]
After the in-depth manual review, LG.

**Pull request merged**